superset/views/chart/views.py

```diff
@@ -68,7 +68,10 @@ def add(self) -> FlaskResponse:
             for d in ConnectorRegistry.get_all_datasources(db.session)
         ]
         payload = {
-            "datasources": sorted(datasources, key=lambda d: d["label"]),
+            "datasources": sorted(
+                datasources,
+                key=lambda d: d["label"].lower() if isinstance(d["label"], str) else "",
```

Comment:
Doing defensive check on `str` in case some old charts in old environments may have a `None` or something